### ISSUE TITLE
The results of the two inferences are inconsistent.

### ISSUE THREAD

**Issue by do1nothing** [Pointcept/Pointcept#281]
Hello, I have encountered an issue during inference. With the same configuration, the prediction results from two inferences are inconsistent, with a small portion of the data having inconsistent labels reaching up to 20%. I noticed that before the data is fed into the model for inference, the coord values differ significantly between the two inferences, although the grid_coord values remain consistent. I am unsure if the coord values would affect the model's inference results since I observed that grid_coord is used during the inference process.

**Comment by RayYoh** [comment]
> Hello, I have encountered an issue during inference. With the same configuration, the prediction results from two inferences are inconsistent, with a small portion of the data having inconsistent labels reaching up to 20%. I noticed that before the data is fed into the model for inference, the coord values differ significantly between the two inferences, although the grid_coord values remain consistent. I am unsure if the coord values would affect the model's inference results since I observed that grid_coord is used during the inference process.

Hi, I encounter same problem as you. Have you find the reasons?

**Comment by do1nothing** [comment]
> > Hello, I have encountered an issue during inference. With the same configuration, the prediction results from two inferences are inconsistent, with a small portion of the data having inconsistent labels reaching up to 20%. I noticed that before the data is fed into the model for inference, the coord values differ significantly between the two inferences, although the grid_coord values remain consistent. I am unsure if the coord values would affect the model's inference results since I observed that grid_coord is used during the inference process.
> 
> Hi, I encounter same problem as you. Have you find the reasons?

yes,I found that the point cloud data is already inconsistent before being fed into the model during inference. The reasons are a random operation occurring during the first GridSample and the introduction of randomness by the four serialization methods. Additionally, during model inference, the sparse convolution in the embedding also causes minor inconsistencies in the output (which I am unable to overcome).

**Comment by Gofinge** [comment]
A slightly difference can not aviod. But the overall performance should be very close.
